app/controllers/concerns/odk_headerable.rb

```diff
@@ -0,0 +1,21 @@
+module OdkHeaderable
+  extend ActiveSupport::Concern
+
+  included do
+    skip_authorize_resource only: :odk_headers
+    before_action :add_openrosa_headers
+
+    def odk_headers
```

Comment:
It's a controller action, for ODK's HEAD and GET requests, I removed the conditionals from the CREATE method, so the routes now respond to HEAD/GET by directing them to `controller#odk_headers`